modules/images-configuration-file.adoc

```diff
@@ -0,0 +1,69 @@
+// Module included in the following assemblies:
+//
+// * openshift_images/image-configuration.adoc
+
+[id="images-configuration-file_{context}"]
+= Configuring image settings
+
+You can configure image registry settings by editing the
+`image.config.openshift.io/cluster` resource.
+
+.Procedure
+
+* Edit the `image.config.openshift.io/cluster` resource:
++
+----
+$ oc edit image.config.openshift.io/cluster
+----
++
+The following is an example `image.config.openshift.io/cluster` resource:
++
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: Image<1>
+metadata:
+  annotations:
+    release.openshift.io/create-only: "true"
+  creationTimestamp: "2019-05-17T13:44:26Z"
+  generation: 1
+  name: cluster
+  resourceVersion: "8302"
+  selfLink: /apis/config.openshift.io/v1/images/cluster
+  uid: e34555da-78a9-11e9-b92b-06d6c7da38dc
+spec:
+  allowedRegistriesForImport:<2>
+    - domainName: quay.io
+      insecure: false
+  additionalTrustedCA:<3>
+    name: myconfigmap
+  registrySources:<4>
+    insecureRegistries:<5>
+    - insecure.com
+    blockedRegistries:<6>
+    - untrusted.com
+status:
+  internalRegistryHostname: image-registry.openshift-image-registry.svc:5000
+----
+<1> `Image`: Holds cluster-wide information about how to handle images. The
+canonical name is `cluster`.
+<2> `allowedRegistriesForImport`: Limits the container image registries from which
+normal users may import images. Set this list to the registries that you trust
+to contain valid Docker images, and that you want applications to be able to
+import from. Users with permission to create images or `ImageStreamMappings`
+from the API are not affected by this policy. Typically only cluster
+administrators will have the appropriate permissions.
+<3> `additionalTrustedCA`: A reference to a ConfigMap containing additional CAs that
+should be trusted during `ImageStream import`, `pod image pull`, and
+`openshift-image-registry pullthrough`. The namespace for this ConfigMap is
+`openshift-config`. The format of the ConfigMap is to use the registry hostname
+as the key, and the base64-encoded certificate as the value, for each additional
+registry CA to trust.
+<4> `registrySources`: Contains configuration that determines how the container
+runtime should treat individual registries when accessing images for builds and
+pods. For instance, whether or not to allow insecure access. It does not contain
+configuration for the internal cluster registry.
+<5> `insecureRegistries`: Registries which do not have a valid SSL certificate or
+only support HTTP connections.
+<6> `blockedRegistries`: Whitelisted for image pull and push actions. All other
+registries are blocked.
```

Comment:
this looks wrong.  blockedRegistries are the blacklisted set, not the whitelisted set.

my example didn't include both "blockdRegistries" and "allowedRegistries" because you can't set both at the same time.  but i think you pasted the description of allowedRegistries here.

Reply:
Thanks! I thought it looked strange. This and the SSL -> TLS might also need to be updated in the API docs.

Reply:
Nevermind, just the SSL -> TLS bit.